src/pages/index.js

```diff
@@ -0,0 +1,333 @@
+  modal.classList.remove("modal_opened");
+}
+
+function renderCard(cardData, placement = "prepend") {
+  const cardElement = getCardElement(cardData);
+
+  if (placement === "append") {
+    cardsList.append(cardElement);
+  } else {
+    cardsList.prepend(cardElement);
+  }
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  const modal__submit = evt.submitter;
+  setButtonText({ btn: modal__submit, isLoading: true });
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      renderCard(cardData, "prepend");
+      closeModal(cardModal);
+      disableButton(cardSubmitBtn, settings);
+      cardFormElement.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText({ btn: modal__submit, isLoading: false });
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const modal__submit = evt.submitter;
+  setButtonText({
+    btn: modal__submit,
+    isLoading: true,
+    defaultText: "Save",
+    loadingText: "Saving...",
+  });
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      jobElement.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText({
+        btn: modal__submit,
+        isLoading: false,
+        defaultText: "Save",
+        loadingText: "Saving...",
+      });
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const modal__submit = evt.submitter;
+  setButtonText({
+    btn: modal__submit,
+    isLoading: true,
+    defaultText: "Save",
+    loadingText: "Saving...",
+  });
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      avatarFormElement.reset();
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText({
+        btn: modal__submit,
+        isLoading: false,
+        defaultText: "Save",
+        loadingText: "Saving...",
+      });
+    });
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = jobElement.textContent;
+  openModal(editModal);
+  resetValidation(
+    editFormElement,
+    [editModalNameInput, editModalDescriptionInput],
+    settings
+  );
+});
+
+// close modal overlay
+modals.forEach((modal) => {
+  modal.addEventListener("click", (evt) => {
+    if (
+      evt.target.classList.contains("modal_opened") ||
+      evt.target.classList.contains("modal__close-btn")
+    ) {
+      closeModal(modal);
+    }
+  });
+});
+
+// click the close button on the edit profile modal to close that modal
+editModalCloseBtn.addEventListener("click", () => {
```

Comment:
Please, delete the redundant `close icon` listeners because you handle them in line 300 **universally**